debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/VirtualThreadsCommand.java

```diff
@@ -0,0 +1,99 @@
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package com.ibm.j9ddr.vm29.tools.ddrinteractive;
+
+import java.io.PrintStream;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodHandles.Lookup;
+import java.lang.invoke.MethodType;
+
+import com.ibm.j9ddr.CorruptDataException;
+import com.ibm.j9ddr.tools.ddrinteractive.Command;
+import com.ibm.j9ddr.tools.ddrinteractive.Context;
+import com.ibm.j9ddr.tools.ddrinteractive.DDRInteractiveCommandException;
+import com.ibm.j9ddr.vm29.j9.DataType;
+import com.ibm.j9ddr.vm29.j9.J9ObjectFieldOffset;
+import com.ibm.j9ddr.vm29.pointer.ObjectReferencePointer;
+import com.ibm.j9ddr.vm29.pointer.PointerPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9JavaVMPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9ObjectPointer;
+import com.ibm.j9ddr.vm29.pointer.helper.J9ObjectHelper;
+import com.ibm.j9ddr.vm29.pointer.helper.J9RASHelper;
+import com.ibm.j9ddr.vm29.types.UDATA;
+
+public class VirtualThreadsCommand extends Command {
+	public VirtualThreadsCommand() {
+		addCommand("vthreads", "", "Lists virtual threads");
+	}
+
+	@Override
+	public void run(String command, String[] args, Context context, PrintStream out) throws DDRInteractiveCommandException {
+		displayVirtualThreads(out);
+	}
+
+	private void displayVirtualThreads(PrintStream out) throws DDRInteractiveCommandException {
+		try {
+			J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
+			if (!JavaVersionHelper.ensureJava19AndUp(vm, out)) {
+				return;
+			}
+			try {
+				Class<J9JavaVMPointer> vmClass = J9JavaVMPointer.class;
+				Lookup lookup = MethodHandles.lookup();
+				MethodHandle liveVirtualThreadListMH = lookup.findVirtual(vmClass, "liveVirtualThreadList", MethodType.methodType(PointerPointer.class));
+				PointerPointer mainVirtualThread = (PointerPointer)liveVirtualThreadListMH.invoke(vm);
+
+				MethodHandle virtualThreadLinkNextOffsetMH = lookup.findVirtual(vmClass, "virtualThreadLinkNextOffset", MethodType.methodType(UDATA.class));
+				UDATA linkNextOffset = (UDATA)virtualThreadLinkNextOffsetMH.invoke(vm);
+
+				if (mainVirtualThread.notNull() && mainVirtualThread.at(0).notNull()) {
+					J9ObjectPointer root = J9ObjectPointer.cast(mainVirtualThread.at(0));
+					J9ObjectPointer node = ObjectReferencePointer.cast(root.addOffset(linkNextOffset)).at(0);
```

Comment:
Can you add a comment describing the structure of `liveVirtualThreadList` and reason for having root/node